test/e2e/framework/pod/exec_util.go

```diff
@@ -141,6 +143,43 @@ func ExecShellInPodWithFullOutput(ctx context.Context, f *framework.Framework, p
 	return execCommandInPodWithFullOutput(ctx, f, podName, "/bin/sh", "-c", cmd)
 }
 
+// VerifyExecInPodSucceed verifies shell cmd in target pod succeed
+func VerifyExecInPodSucceed(ctx context.Context, f *framework.Framework, pod *v1.Pod, shExec string) {
```

Comment:
Helper functions in `test/e2e/framework/*` packages should return an error instead of recording a test failure.

That makes them more reusable, allows callers to add more information for an error, and last but not least makes failures point to the test instead of the helper.

Reply:
@pohly updated. please review it again.